superset/superset/db_engine_specs.py

```diff
@@ -1152,7 +1152,22 @@ def fetch_data(cls, cursor, limit):
         try:
             return super(HiveEngineSpec, cls).fetch_data(cursor, limit)
         except pyhive.exc.ProgrammingError:
+            logging.warning('An exception occured while fetching data, '
+                            'returning empty result')
             return []
+        except pyhive.exc.OperationalError as op_err:
+            # The `pyhive.exc.OperationalError` exception is raised when a
+            # query is cancelled. This seems to happen because `gsc` expects
+            # the state to be `FINISHED` but got `CANCELED`.
+            cancelation_error_msg = 'Expected state FINISHED, but found CANCELED'
```

Comment:
please add FIXME comment as it's needed only while gsc doesn't support CANCELED.

Reply:
Done [here](https://github.com/src-d/sourced-ui/pull/267/commits/0c06f07bf393d97a9f2f5c2490c1de03bf9ddc84).